src/docs/arch-row.dox

```diff
@@ -1,12 +1,68 @@
-/*! @arch_page arch-row Row Store
+/*! @arch_page arch-row-column Row Store & Column Store
 
-Row Stores are Btrees stored in WiredTiger that do not have a record id
-as key.  Thus, they implement a generalized version of a Btree, where
-the key and data can be arbitrary length.
+@section Definition
+
+A row oriented storage is a database management system where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column, the data is stored by column. WriedTiger supports both and each management system is
+represented by a certain type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. However, column-stores use a 64-bit unsigned integer record id as
+keys, thus, column-stores implement a specialized version of a B-Tree where the key has a
+predictable length.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+ is created. The fixed length column store has specialized use cases like bitmaps
+- The variable length is a more general case which allows for values to have any length, and may
+have arbitrary types, including aggregates of various types.
+
+@section Main differences
+
+Column-store keys are derived from the value's location in the table while row-store keys are
+physically stored in the table. This means there is no repetition of keys across multiple files for
+column-stores while row-store keys are duplicated usually within the same single file. Therefore,
+column-stores have a real benefit over row-stores when it comes to finding a record, since there is
+only one key per page and keys take almost no storage, fast lookups can be performed and it is easy
+to move between records. For row-stores, a slower approach is necessary (i.e multiple binary
+searches) to locate a particular record and ultimately pull together a complete row.
```

Comment:
What are you trying to get at with "no repetition of keys across multiple files"? You mean, indexes? Or something else?

I don't understand "one key per page", unless you mean there's  a "starting key" on each page and that's used to figure out the rest of the keys on the page?

I'm not sure why it's easier to move between records in CS vs. RS, I'm not sure what you're trying to say?

Are you trying to talk about internal pages vs. leaf pages?

Finally, RLE in CS implies that any single record can take up a large number of record numbers, so there are some interesting search problems, you can't simply jump to a record in a leaf page.

Reply:
> What are you trying to get at with "no repetition of keys across multiple files"? You mean, indexes? Or something else?
I probably got confused with what's [here](https://github.com/wiredtiger/wiredtiger/blob/develop/src/docs/schema.dox#L247).

> I don't understand "one key per page", unless you mean there's a "starting key" on each page and that's used to figure out the rest of the keys on the page?
I am not sure to get how this is done, I need to understand before being able to write about it.

> I'm not sure why it's easier to move between records in CS vs. RS, I'm not sure what you're trying to say?
I thought that finding a specific record would be easier with CS than with RS, but after thinking about, it's probably not true.

> Are you trying to talk about internal pages vs. leaf pages?
I did not want to get into that level of detail, maybe I should ? It seems to be covered here: #6816.

> Finally, RLE in CS implies that any single record can take up a large number of record numbers, so there are some interesting search problems, you can't simply jump to a record in a leaf page.
That is interesting, I could add it to the page.